pydantic/_internal/_forward_ref.py

```diff
@@ -48,7 +49,7 @@ class PydanticForwardRef:
     """
 
     schema: core_schema.CoreSchema
-    model: type[BaseModel]
+    model: type[BaseModel] | type[PydanticDataclass]
```

Comment:
any reason not to change this to a type[Any]? I can see these are basically the only types we use it on today, but I don't see any of the APIs depending on this being one of those two types. I might be missing something though.

Reply:
Let's keep it stricter for now, easy to relax it - maybe add a comment to that effect.